libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.java

```diff
@@ -541,8 +569,37 @@ public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request
                         onAuthFlowComplete(tr);
                     }
                 }
+            } else {
+                // Only allow redirect for requests with authorization headers for Salesforce urls.
+                if (request.getRequestHeaders() != null && request.getRequestHeaders().containsKey("Authorization")) {
+                   return isSalesforceUrl(formattedUrl);
+                }
             }
-            return isDone;
+
+            return authFlowFinished;
+        }
+
+        // List from https://help.salesforce.com/s/articleView?language=en_US&id=sf.domain_name_url_formats.htm&type=5
+        private boolean isSalesforceUrl(String host) {
+            return (host != null && (host.endsWith(".salesforce.com") ||
+                    host.endsWith(".force.com") ||
+                    host.endsWith(".sfdcopens.com") ||
+                    host.endsWith(".site.com") ||
+                    host.endsWith(".lightning.com") ||
+                    host.endsWith(".salesforce-sites.com") ||
+                    host.endsWith(".force-user-content.com") ||
+                    host.endsWith(".salesforce-experience.com") ||
+                    host.endsWith(".salesforce-scrt.com")));
+        }
+
+        private boolean isNewLoginUrl(Uri uri) {
+            if (!uri.toString().contains(getLoginUrl()) && isSalesforceUrl(uri.getHost())) {
```

Comment:
I only added support for *.my.salesforce.com on iOS, do we want to open it up to more?

Reply:
@bbirman but you have that logic in SFNetwork for regular rest calls not during login, right?

Reply:
The login one just checks for `.my.salesforce.com/?startURL=/setup/secur/RemoteAccessAuthorizationPage.apexp` (draft incoming), and then the other logic to check for Salesforce URLs and conditionally redirect is just in SFNetwork like you said

Reply:
Good callout, it should be more narrow.  Is that the only domain though?  What about community url (`*.force.com`)?

Reply:
I feel like the UI nudges towards just adding a prefix to .my.salesforce.com so I've been on the fence about it, do either of you have a preference? is the startURL portion the same for communities too?